### PR TITLE
Jquery resize

### PR DESCRIPTION
**Pull request by ayumihamsaki**
See github here: https://github.com/octobercms/october/pull/4461

### CODE REVIEW

**Comment by ayumihamsaki** [comment]
Finished.

File: modules/system/assets/js/mailbrandsettings/mailbrandsettings.js [hunk 1]
```diff
@@ -33,5 +33,7 @@ function createPreviewContainer(el, content) {
      * Auto adjust height
      */
     $(document).render(adjustPreviewHeight)
-    $(window).resize(adjustPreviewHeight)
+    $(window).on('resize', function() {
```
Comment (w20k) [inline modules/system/assets/js/mailbrandsettings/mailbrandsettings.js]: @ayumihamsaki I think you could simplify.
```suggestion
    $(window).on('resize', adjustPreviewHeight);
```

File: themes/demo/assets/vendor/bootstrap/js/modal.js [hunk 2]
```diff
@@ -158,13 +158,13 @@
     }
   }
 
-  Modal.prototype.resize = function () {
+  Modal.prototype.on('resize', function() {
```
Comment (w20k) [inline themes/demo/assets/vendor/bootstrap/js/modal.js]: @ayumihamsaki revert this one.

File: modules/backend/formwidgets/codeeditor/assets/js/codeeditor.js [hunk 3]
```diff
@@ -285,7 +285,7 @@
     }
 
     CodeEditor.prototype.onResize = function() {
-        this.editor.resize()
+        this.editor.trigger('resize')
```
Comment (w20k) [inline modules/backend/formwidgets/codeeditor/assets/js/codeeditor.js]: @ayumihamsaki I guess method that's called resize of the editor, not the winodw resize (but need to double-check).

File: modules/system/assets/js/mailbrandsettings/mailbrandsettings.js (hunk 1 above)
Comment (ayumihamsaki) [reply modules/system/assets/js/mailbrandsettings/mailbrandsettings.js]: done

File: themes/demo/assets/vendor/bootstrap/js/modal.js (hunk 2 above)
Comment (ayumihamsaki) [reply themes/demo/assets/vendor/bootstrap/js/modal.js]: done

File: modules/backend/formwidgets/codeeditor/assets/js/codeeditor.js (hunk 3 above)
Comment (ayumihamsaki) [reply modules/backend/formwidgets/codeeditor/assets/js/codeeditor.js]: @w20k I've wait for you to confirm or not?

Comment (w20k) [reply modules/backend/formwidgets/codeeditor/assets/js/codeeditor.js]: Yeah, will confirm and post a comment ;) Thanks!

**Comment by ayumihamsaki** [comment]
## Add debounce to .resize()

List of files containing resize:

- [ ] https://github.com/octobercms/october/blob/master/modules/system/assets/ui/vendor/flot/jquery.flot.resize.js

- [ ] https://github.com/octobercms/october/blob/master/modules/backend/formwidgets/codeeditor/assets/js/codeeditor.js

- [ ] https://github.com/octobercms/october/blob/master/modules/backend/assets/js/october.sidepaneltab.js

- [ ] https://github.com/octobercms/october/blob/master/themes/demo/assets/vendor/bootstrap/js/modal.js

- [ ] https://github.com/octobercms/october/blob/master/modules/backend/formwidgets/markdowneditor/assets/js/markdowneditor.js

- [ ] https://github.com/octobercms/october/blob/master/modules/backend/assets/js/october.filelist.js

- [ ] https://github.com/octobercms/october/blob/master/modules/system/assets/js/mailbrandsettings/mailbrandsettings.js

- [ ] https://github.com/octobercms/october/blob/master/modules/backend/assets/js/october.layout.js

- [ ] https://github.com/octobercms/october/blob/master/modules/system/assets/ui/js/toolbar.js

~ To do

File: modules/backend/formwidgets/codeeditor/assets/js/codeeditor.js (hunk 3 above)
Comment (w20k) [reply modules/backend/formwidgets/codeeditor/assets/js/codeeditor.js]: @ayumihamsaki I was correct, it uses the ace.js resize method:
Usage:
https://github.com/octobercms/october/blob/master/modules/backend/formwidgets/codeeditor/assets/vendor/ace/ace.js#L11988
Implementation:
https://github.com/octobercms/october/blob/master/modules/backend/formwidgets/codeeditor/assets/vendor/ace/ace.js#L15723

Could you revert all the expressions back to the old version: `this.editor.resize()`

Comment (ayumihamsaki) [reply modules/backend/formwidgets/codeeditor/assets/js/codeeditor.js]: revert whole file

File: modules/backend/formwidgets/markdowneditor/assets/js/markdowneditor.js [hunk 4]
```diff
@@ -156,7 +156,7 @@
     }
 
     MarkdownEditor.prototype.onResize = function() {
-        this.editor.resize()
```
Comment (w20k) [inline modules/backend/formwidgets/markdowneditor/assets/js/markdowneditor.js]: @ayumihamsaki everything inside of this file as well :)

File: modules/system/assets/ui/vendor/flot/jquery.flot.resize.js [hunk 5]
```diff
@@ -41,20 +41,20 @@ can just fix the size of their placeholders.
             if (placeholder.width() == 0 || placeholder.height() == 0)
                 return;
 
-            plot.resize();
+            plot.trigger('resize');
```
Comment (w20k) [inline modules/system/assets/ui/vendor/flot/jquery.flot.resize.js]: Not needed here, I think.

File: themes/demo/assets/vendor/bootstrap/js/modal.js [hunk 6]
```diff
@@ -63,7 +63,7 @@
     this.$body.addClass('modal-open')
 
     this.escape()
-    this.resize()
```
Comment (w20k) [inline themes/demo/assets/vendor/bootstrap/js/modal.js]: @ayumihamsaki model.js has its own resize method, so not needed here as well:
https://github.com/octobercms/october/blob/fc3c4de9070f2b1b1ab95fe83390ce92d6b0a17c/themes/demo/assets/vendor/bootstrap/js/modal.js#L161

File: modules/backend/formwidgets/markdowneditor/assets/js/markdowneditor.js (hunk 4 above)
Comment (ayumihamsaki) [reply modules/backend/formwidgets/markdowneditor/assets/js/markdowneditor.js]: yah was thinking that as well on that one

File: modules/system/assets/ui/vendor/flot/jquery.flot.resize.js (hunk 5 above)
Comment (ayumihamsaki) [reply modules/system/assets/ui/vendor/flot/jquery.flot.resize.js]: done

File: themes/demo/assets/vendor/bootstrap/js/modal.js (hunk 6 above)
Comment (ayumihamsaki) [reply themes/demo/assets/vendor/bootstrap/js/modal.js]: done

File: modules/system/assets/ui/js/jquery.debouncedresize.js.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
+/*
+ * debouncedresize: special jQuery event that happens once after a window resize
```
Comment (w20k) [inline modules/system/assets/ui/js/jquery.debouncedresize.js.js]: @ayumihamsaki  double `.js.js` in the filename - `jquery.debouncedresize.js.js`

Comment (ayumihamsaki) [reply modules/system/assets/ui/js/jquery.debouncedresize.js.js]: thanks

**Comment by ayumihamsaki** [comment]
ok I think that's done - sorry about so many commits, github is caching everything and it's driving me crazy! It's not showing me the correct screen 99% of the time!

**Comment by ayumihamsaki** [comment]
Closed please see the main jQuery update PR.

**Pull request closed without merging**